src/ripple/app/paths/Pathfinder.cpp

```diff
@@ -1185,8 +1185,10 @@ void fillPaths (Pathfinder::PaymentType type, PathCostList const& costs)
 void Pathfinder::initPathTable ()
 {
     // CAUTION: Do not include rules that build default paths
-    fillPaths(
-        pt_XRP_to_XRP, {});
+
+    // When the app is destroyed and a new app is created, start with a fresh mPathTable
+    mPathTable.clear();
+    fillPaths (pt_XRP_to_XRP, {});
```

Comment:
Regardless of where it is, does the comment really add anything? I guess it's explaining *why* you're clearing `mPathTable`, so nobody takes the line out later, but that'll probably go stale pretty quickly as stopping and starting the app becomes more common.